tools/compatibility-matrix-automation/src/matrix-automation.ts

```diff
@@ -0,0 +1,142 @@
+import { readFileSync, readdirSync, writeFileSync } from "fs";
+import { join, extname } from "path";
+import { TestResultsJson } from "winglang/dist/commands/test/results";
+import { CompatibilityMatrix, CompatibilitySets } from "./types";
+
+export const SKIPPED_RESOURCES = ["TestRunner"];
+export const MATRIX_PATH = join(
+  __dirname,
+  "../../../docs/docs/04-standard-library/compatibility/compatibility.json"
+);
+export const OUT_PATH = join(__dirname, "../../../out.json");
+
+export const PLATFORMS = {
+  "tf-aws": "tf-aws",
+  "tf-gcp": "tf-gcp",
+  "tf-azure": "tf-azure",
+  awscdk: "awscdk", // in the future we plan to move awscdk to its own compatibility matrix https://github.com/winglang/wing/issues/5560
+  sim: "sim",
+};
+
+export function updateMatrix(outFolderPath: string, matrixPath = MATRIX_PATH) {
+  const files = readdirSync(join(outFolderPath), {
+    withFileTypes: true,
+  }).filter((file) => file.isFile() && extname(file.name) === ".json");
+
+  const matrix = JSON.parse(
+    readFileSync(matrixPath).toString()
+  ) as CompatibilityMatrix;
+
+  for (const file of files) {
+    try {
+      const outFile = JSON.parse(
+        readFileSync(join(file.path, file.name), "utf8")
+      ) as TestResultsJson;
+
+      updateMatrixFromFile(outFile, matrix);
+      console.log(`"${file.name}" parsed successfully!`);
+    } catch (error) {
+      console.error(
+        `an error occurred while trying to parse "${file.name}". error: ${error}`
+      );
+    }
+  }
+
+  writeFileSync(MATRIX_PATH, JSON.stringify(matrix, null, 2));
+}
+
+export function updateMatrixFromFile(
+  outFile: TestResultsJson,
+  matrix: CompatibilityMatrix
+) {
+  const compatibility: Map<string, CompatibilitySets> = new Map();
+
+  const platform = outFile.platforms[0] as keyof typeof PLATFORMS;
+
+  for (const testFile of Object.values(outFile.results)) {
+    for (const testName in testFile) {
+      const {
+        pass,
+        unsupported,
+        unsupportedOperation,
+        unsupportedResource,
+        args = {},
+      } = testFile[testName];
+      if (unsupported && unsupportedOperation && unsupportedResource) {
+        addToCompatibilitySet(
+          compatibility,
+          unsupportedResource,
+          unsupportedOperation,
+          false
+        );
+      } else {
+        if (args.methods) {
+          for (const resource in args.methods) {
+            if (SKIPPED_RESOURCES.includes(resource)) {
+              continue;
+            }
+            (args.methods as Record<string, string[]>)[resource].forEach((op) =>
```

Comment:
Is the casting here needed? Doesn't this have types from `TestResultsJson`?

Reply:
args is not modeled (since it's not really part of wing but added on a separate platform)